engine/access/rest/handlers.go

```diff
@@ -0,0 +1,228 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const BlockIDCntLimit = 50
+
+var MaxAllowedBlockIDsCnt = BlockIDCntLimit
+
+// Handlers provide collection of handlers used by the API server
+type Handlers struct {
+	backend access.API
+	logger  zerolog.Logger
+}
+
+func NewHandlers(backend access.API, logger zerolog.Logger) *Handlers {
+	return &Handlers{
+		backend: backend,
+		logger:  logger,
+	}
+}
+
+func (h *Handlers) BlocksIdGet(w http.ResponseWriter, r *http.Request) {
+	// create h logger for the request
+	errorLogger := h.logger.With().Str("request_url", r.URL.String()).Logger()
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	vars := mux.Vars(r)
+	idParam := vars["id"]
+
+	// gorilla mux retains opening and ending square brackets for ids
+	idParam = strings.TrimSuffix(idParam, "]")
+	idParam = strings.TrimPrefix(idParam, "[")
+
+	ids := strings.Fields(idParam)
```

Comment:
looking into this now - The generated Rest client does a `fmt.Sprintf("%v", id)` for the id which is of type `[]string` resulting in a space separated list of ids instead of a comma separated one.